sockets.go

```diff
@@ -157,6 +157,49 @@ type JSONConnection struct {
 	Receiver reflect.Value
 }
 
+// Binary Connection connects a websocket message connection to a []byte
+// channel.
+type BinaryConnection struct {
+	*Connection
+
+	// Sender is the []byte channel used for sending out binary data to the client.
+	// This channel gets mapped for the next handler to use and is asynchronous
+	// unless the SendChannelBuffer is set to 0.
+	Sender chan []byte
+
+	// Receiver is the []byte channel used for receiving binary data from the client.
```

Comment:
Similarly here this is not specific to receiving binary data. I know the motivation for this is protobufs, but technically this is agnostic to what the byte slice represents.